apps/meteor/app/livechat/server/hooks/afterAgentRemoved.ts

```diff
@@ -0,0 +1,13 @@
+import { LivechatDepartment, Users, LivechatDepartmentAgents, LivechatVisitors } from '@rocket.chat/models';
+
+import { callbacks } from '../../../../lib/callbacks';
+
+callbacks.add('livechat.afterAgentRemoved', async ({ agent }) => {
+	const departmentIds = (await LivechatDepartmentAgents.findByAgentId(agent._id).toArray()).map((department) => department.departmentId);
+	await Promise.all([
+		Users.removeAgent(agent._id),
+		LivechatDepartmentAgents.removeByAgentId(agent._id),
+		agent.username && LivechatVisitors.removeContactManagerByUsername(agent.username),
```

Comment:
I think this might be a problem since I don't see username prop on the projection of this query